spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/archaius/ArchaiusAutoConfiguration.java

```diff
@@ -165,14 +158,30 @@ protected void configureArchaius(ConfigurableEnvironmentConfiguration envConfig)
 			config.setContainerConfigurationIndex(config
 					.getIndexOfConfiguration(appOverrideConfig));
 
-			ConfigurationManager.install(config);
-		}
-		else {
+			addArchaiusConfiguration(config);
+		} else {
 			// TODO: reinstall ConfigurationManager
 			log.warn("Netflix ConfigurationManager has already been installed, unable to re-install");
 		}
 	}
 
+	private void addArchaiusConfiguration(ConcurrentCompositeConfiguration config) {
```

Comment:
In your test you use `ConfigurationManager.install(abstractConfiguration);`.  I would rather inject a Collection of configuration beans and add them in this class rather than relying on the archaius static methods.